### PR TITLE
Add a test for ROCm libraries and build rules

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

This PR adds new packages to test the implementation and usage of ROCm libraries for device code in CMSSW plugins and executables.

The package `HeterogeneousTest/ROCmDevice` implements a library that defines and exports ROCm device-only functions, and a plugin and test that use them.

The package `HeterogeneousTest/ROCmKernel` implements a library that imports device functions from `HeterogeneousTest/ROCmDevice` to define and export ROCm kernels, and a plugin and test that use them.

The package `HeterogeneousTest/ROCmWrapper` implements a library that imports kernels from `HeterogeneousCTest/ROCmKernel` to define and export host-only wrappers around them, usable by non-ROCm libraries, plugins and applications, and implements a plugin and test that use them.

The package `HeterogeneousTest/ROCmOpaque` implements a library that use the wrappers from `HeterogeneousTest/ROCmWrapper` to define and export host-only functions around the whole ROCm section, usable by libraries, plugins and applications that are not ROCm-aware, and implements a plugin and test that use them.

**The PR builds on top of and includes**:
  - #40619
  - #40642

#### PR validation:

The new unit tests compile and pass.

### CODE REVIEW

File: HeterogeneousTest/ROCmOpaque/test/testROCmTestAdditionModules.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+process.rocmTestWrapperAdditionModule = cms.EDAnalyzer('ROCmTestWrapperAdditionModule',
+    size = cms.uint32( 1024*1024 )
+)
+
+process.rocmTestOpaqueAdditionModule = cms.EDAnalyzer('ROCmTestOpaqueAdditionModule',
+    size = cms.uint32( 1024*1024 )
+)
+
+process.path = cms.Path(
+     process.rocmTestDeviceAdditionModule +
```
Comment (fwyzard) [inline HeterogeneousTest/ROCmOpaque/test/testROCmTestAdditionModules.py:26]: @makortel note that this one works with ROCm, while it does not work with CUDA.

**Comment by fwyzard** [comment]
please test with cms-sw/cmsdist#8273,#40619,#40642

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
+heterogeneous

**Comment by fwyzard** [comment]
code-checks with cms.week0.PR_66839164/53.0-f55d3c97efc0e8731e06c12170e029af

File: HeterogeneousTest/ROCmOpaque/test/testROCmTestAdditionModules.py (hunk 1 above)
Comment (makortel) [reply HeterogeneousTest/ROCmOpaque/test/testROCmTestAdditionModules.py:26]: Nice

**Comment by fwyzard** [comment]
please test with cms-sw/cmsdist#8273

**Comment by fwyzard** [comment]
code-checks with cms.week0.PR_66839164/53.0-f55d3c97efc0e8731e06c12170e029af

**Comment by fwyzard** [comment]
please test with cms-sw/cmsdist#8273

**Comment by fwyzard** [comment]
code-checks with cms.week0.PR_66839164/53.0-f55d3c97efc0e8731e06c12170e029af

**Comment by fwyzard** [comment]
please test with cms-sw/cmsdist#8273

**Comment by fwyzard** [comment]
hold

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
+heterogeneous

**Comment by fwyzard** [comment]
unhold

**Comment by perrotta** [comment]
+1
- Updates in HeterogeneousCore/ROCmServices were already merged with #40642
- Updates in HeterogeneousTest are specific of this PR

**Comment by perrotta** [comment]
merge

**Pull request merged**